storm-client/src/jvm/org/apache/storm/utils/ConfigUtils.java

```diff
@@ -313,6 +315,34 @@ public static File getWorkerDirFromRoot(String logRoot, String id, Integer port)
         return new File((logRoot + FILE_SEPARATOR + id + FILE_SEPARATOR + port));
     }
 
+    /**
+     * Get the given config value as a String, if possible.
+     * @param name - the config key
+     * @param conf - the config map
+     * @return - the config value converted to a String if found, otherwise null.
+     * @throws IllegalArgumentException if conf is null
+     * @throws NullPointerException if name is null and the conf map doesn't support null keys
+     */
+    public static String getConfigValueAsString(String name, Map<?, ?> conf) {
```

Comment:
@revans2 You mean, replace this method with
```
public static String getConfigValueAsList(String name, Map<?, ?> conf) {
    // if the value is a list, return it
    // if the value is a string, split it by space and return the resulting list
}
```
and the caller can convert it to a space-separated string if needed?